packages/generators/app/lib/utils/check-requirements.js

```diff
@@ -1,14 +1,28 @@
 'use strict';
 
+const { red, green, bold, yellow } = require('chalk');
+const semver = require('semver');
+const packageJSON = require('../resources/json/common/package.json');
+
 module.exports = function checkBeforeInstall() {
   const currentNodeVersion = process.versions.node;
-  const semver = currentNodeVersion.split('.');
-  const major = semver[0];
+  const { engines } = packageJSON({ strapiDependencies: [] });
 
-  if (major < 12) {
-    console.error(`You are running Node ${currentNodeVersion}`);
-    console.error('Strapi requires Node 12 and higher.');
+  // error if the node version isn't supported
+  if (!semver.satisfies(currentNodeVersion, engines.node)) {
+    console.error(red(`You are running ${bold(`Node.js ${currentNodeVersion}`)}`));
+    console.error(`Strapi requires ${bold(green(`Node.js ${engines.node}`))}`);
     console.error('Please make sure to use the right version of Node.');
     process.exit(1);
   }
+
+  // warn if not using a LTS version
+  else if (!semver.satisfies(currentNodeVersion, '14.x.x || 16.x.x || 18.x.x || 20.x.x')) {
```

Comment:
This way 18 and 20 will automatically start working fine once official support is added in generated package.json without needing to update this file.